internal/controller/postgrescluster/postgres.go

```diff
@@ -375,6 +376,36 @@ func (r *Reconciler) reconcilePostgresUserSecrets(
 			))
 	}
 
+	// Sorts the slice of secrets.Items based on secrets with identical labels
+	// If one secret has "pguser" in its name and the other does not, the
+	// one without "pguser" is moved to the front.
+	// If both secrets have "pguser" in their names or neither has "pguser", they
+	// are sorted by creation timestamp.
+	// If two secrets have the same creation timestamp, they are further sorted by name.
+	// The secret to be used by PGO is put at the end of the sorted slice.
+	sort.Slice(secrets.Items, func(i, j int) bool {
```

Comment:
🌱 This comparison function only uses `secrets.Items`, so it could be separated into another function if you want to test that. Something like:

```go
func sortPostgresUserSecrets(secrets []corev1.Secret) {
  sort.Slice(secrets, func(i, j int) bool {
    …
  })
}
```